From fee59f195192983863ae0990a8b7ca101a807ccd Mon Sep 17 00:00:00 2001
From: Timm Allman <tallman@umass.edu>
Date: Thu, 1 Oct 2015 21:39:56 -0400
Subject: [PATCH] Add a way to destinguish HL mmap calls from application mmap
 calls

---
 heaps/top/mmapheap.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/heaps/top/mmapheap.h b/heaps/top/mmapheap.h
index cff0129..d74239c 100644
--- a/heaps/top/mmapheap.h
+++ b/heaps/top/mmapheap.h
@@ -122,7 +122,7 @@ namespace HL {
       static int fd = ::open ("/dev/zero", O_RDWR);
       void * ptr = mmap (NULL, sz, HL_MMAP_PROTECTION_MASK, MAP_PRIVATE, fd, 0);
 #else
-      void * ptr = mmap (NULL, sz, HL_MMAP_PROTECTION_MASK, MAP_PRIVATE | MAP_ANONYMOUS, -1, 0);
+      void * ptr = mmap (NULL, sz, HL_MMAP_PROTECTION_MASK, MAP_PRIVATE | MAP_ANONYMOUS | 0x100000, -1, 0);
 #endif
       if (ptr == MAP_FAILED) {
 	ptr = NULL;
-- 
2.1.0

